snuba/query/query_settings.py

```diff
@@ -0,0 +1,32 @@
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from typing import List
+
+from snuba.state.quota import ResourceQuota
+from snuba.state.rate_limit import RateLimitParameters
+
+
+@dataclass()
```

Comment:
Design question. The original intent of the RequestSettings object (which seems to me is being replaced by QuerySettings) was to keep it immutable (and then, obviously, we violated our own architecture and added the mutable rate limiter list but regardless ...). The idea was that it would only contain settings provided by the user, thus not modified by the query processing pipeline.

Do you plan to make `QuerySettings` immutable or do you expect the query pipeline could make changes to it ?